buildrunner/docker/multiplatform_image_builder.py

```diff
@@ -212,6 +216,46 @@ def _stop_local_registry(self):
         else:
             logger.warning("Local registry is not running when attempting to stop it")
 
+    # pylint: disable=too-many-arguments,too-many-locals
+    def _build_with_inject(self,
+                           inject: dict,
+                           tagged_names: List[str],
+                           platform: str,
+                           push: bool,
+                           path: str,
+                           file: str,
+                           build_args: dict,) -> None:
+
+        if not path or not os.path.isdir(path):
+            logger.warning(f"Failed to inject {inject} for {tagged_names} since path {path} isn't a directory.")
+            return
+
+        dir_prefix = 'mp-tmp-dir'
+        with tempfile.TemporaryDirectory(dir=self._temp_dir, prefix=dir_prefix) as tmp_dir:
```

Comment:
I'm a little confused why we're creating a temporary directory within a temporary directory. If I understand the code, it'll look something like this: `{temp_dir}/mp-tmp-dir-{random-string}/mp-tmp-dir`. Why not just use the `{temp_dir}/mp-tmp-dir-{random-string}` dir for the directory to use instead of creating another temporary directory underneath? Or maybe shutil.copytree requires copying into a new directory like that?

Reply:
Good catch. I made a bad assumption about what get_temp_dir() did and thought it was just a location to create a temp dir and not a actual tempfile.temp_dir(). I'll fix that in a another PR.

Reply:
The dest path can't already exist with shutil.copytree() that is why I appended mp-tmp-dir at the end.

Reply:
Based on your conversation I will leave this as is. Since it gives full control over when these copied files get deleted to ensure that it gets cleaned up.